### PR TITLE
[BUILD] Enable .asf.yaml features

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

Follow https://cwiki.apache.org/confluence/display/INFRA/git+-+.asf.yaml+features to enable `.asf.yaml` with some Github features
- desc
- homepage
- tags/labels
- merge buttons - squash only
- set master as protected, which need at least 1 approval before getting merged
- enabling Github issuses
- notifications w/ mailing lists

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @WillemJiang @Apache9  @aajisaka  @zjffdu @pan3793 @turboFei @ulysses-you @jhx1008 thanks

File: .asf.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - hadoop
+    - hive
+    - sql
+    - kubernetes
+  enabled_merge_buttons:
+    squash: true
+    merge: false
+    rebase: false
+  protected_branches:
+    master:
```
Comment (pan3793) [inline .asf.yaml:38]: This strategy may break our merge script

Comment (yaooqinn) [reply .asf.yaml:38]: we can try first

File: .asf.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# See: https://cwiki.apache.org/confluence/display/INFRA/git+-+.asf.yaml+features
+
+github:
+  description: "Apache Kyuubi is a distributed multi-tenant JDBC server for large-scale data processing and analytics, built on top of Apache Spark"
```
Comment (pan3793) [inline .asf.yaml:21]: I'm not sure if the **Apache Kyuubi** without **(Incubating)** meets the Apache trademark requirements.

Comment (yaooqinn) [reply .asf.yaml:21]: These rules are generalized to describe ASF software associated with the trademark "Apache ProjectName", or more generally "ProjectName" when it is understood to refer to our specific Apache ProjectName software product. Like most ASF software, this ProjectName software is maintained by the Apache ProjectName project, or by the ProjectName sub-project of another project, such as the "Apache Incubator" (itself an ASF trademark).

https://incubator.apache.org/projects/

**Review by pan3793 — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**